### PR TITLE
Delete all table data when cleaning up dangling cells

### PR DESCRIPTION
**Pull request by maackle**
### Summary

No data should remain when deleting a dangling cell

### TODO:
- [x] CHANGELOGs updated with appropriate info
- [x] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/holochain/src/conductor/conductor.rs [hunk 1]
```diff
@@ -2816,57 +2819,51 @@ impl Conductor {
...
             .iter()
             .map(|cell| cell.id().dna_hash())
             .filter(|dna| !all_dnas.contains(dna));
 
-        // For any unrepresented DNAs, clean up those DNA-specific databases
-        for dna_hash in dnas_to_cleanup {
+        // For any unrepresented DNAs, purge data from those DNA-specific databases
+        for dna_hash in dnas_to_purge {
+            // Delete all data from authored databases
+            // TODO: we should delete the entire database directory, since these databases
```
Comment (jost-s) [inline crates/holochain/src/conductor/conductor.rs:2835]: There is the `.path()` of a database could be used to delete the database file `std::fs::remove_file(db.path())`. What do you think about doing that?

Comment (ThetaSinner) [reply crates/holochain/src/conductor/conductor.rs:2835]: Agree, prefer removing the file. All the database files are in one directory, it's flat - so removing the file sounds like the best option to me

Comment (maackle) [reply crates/holochain/src/conductor/conductor.rs:2835]: I wonder if the connection pool needs to be cleaned up too then. I'll write a test if there isn't one already.

File: crates/holochain/src/conductor/conductor.rs [hunk 2]
```diff
@@ -2816,57 +2819,77 @@ impl Conductor {
...
+                            path.set_extension("");
+                            let stem = path.to_string_lossy();
+                            for ext in ["db-shm", "db-wal"] {
+                                let path = PathBuf::from(format!("{stem}.{ext}"));
+                                if let Err(err) = ffs::remove_file(&path).await {
+                                    let err = err.remove_backtrace();
+                                    tracing::warn!(?err, "Failed to remove DB file");
+                                }
+                            }
+                            // db.write_async(purge_data).await?;
```
Comment (jost-s) [inline crates/holochain/src/conductor/conductor.rs:2864]:
```suggestion
```

File: crates/holochain/src/conductor/conductor/tests.rs [hunk 3]
```diff
@@ -294,6 +294,13 @@ async fn test_uninstall_app() {
         (2, 0)
     );
 
+    let dbs = conductor
+        .spaces
+        .get_all_authored_dbs(dna.dna_hash())
+        .unwrap();
+    std::fs::File::open(dbs[0].path()).unwrap();
+    std::fs::File::open(dbs[1].path()).unwrap();
+
```
Comment (jost-s) [inline crates/holochain/src/conductor/conductor/tests.rs:303]: I suggest to store the paths in variables and check at the end of the test that at least the `sqlite3` files are gone from the file system.

**Review by jost-s — APPROVED** [review]
Looking great!

**Pull request merged**